public/src/components/channelManagement/bylineWithImageEditor.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useEffect } from 'react';
+import { Checkbox, makeStyles, TextField, Theme } from '@material-ui/core';
+import { useForm } from 'react-hook-form';
+import { EMPTY_ERROR_HELPER_TEXT } from './helpers/validation';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import { BylineWithImage } from './helpers/shared';
+
+const useStyles = makeStyles(({ spacing }: Theme) => ({
+  container: {
+    marginBottom: spacing(3),
+  },
+}));
+
+interface BylineWithImageEditorProps {
+  bylineWithImage: BylineWithImage;
+  updateBylineWithImage: (bylineWithImage: BylineWithImage) => void;
+  isDisabled: boolean;
+  onValidationChange: (isValid: boolean) => void;
+}
+
+const DEFAULT_BYLINE: BylineWithImage = {
+  name: '',
+};
+
+const BylineWithImageEditor: React.FC<BylineWithImageEditorProps> = ({
+  bylineWithImage,
+  updateBylineWithImage,
+  isDisabled,
+  onValidationChange,
+}: BylineWithImageEditorProps) => {
+  const generateDefaultValues = () => {
+    const obj: BylineWithImage = {
+      name: bylineWithImage.name || '',
+    };
+    if (bylineWithImage.description != null) {
+      obj.description = bylineWithImage.description;
+    }
+    if (bylineWithImage.headshot != null) {
+      obj.headshot = {
+        mainUrl: bylineWithImage.headshot.mainUrl,
+        altText: bylineWithImage.headshot.altText,
+      };
+    }
+    return obj;
+  };
+  const defaultValues: BylineWithImage = generateDefaultValues();
+
+  const modifyReturnCheck = () => {
+    if (bylineWithImage != null && bylineWithImage.headshot != null) {
+      if (!bylineWithImage.headshot.mainUrl && !bylineWithImage.headshot.altText) {
+        delete bylineWithImage.headshot;
+      }
+    }
+    if (defaultValues != null && defaultValues.headshot != null) {
```

Comment:
I don't think we want to be accessing `defaultValues` here, I'm going to suggest an alternative approach using `react-hook-form` more correctly

Reply:
how's this? https://github.com/guardian/support-admin-console/pull/416

Reply:
Alternative approach merged